spec/solver_spec.rb

```diff
@@ -0,0 +1,43 @@
+require_relative '../solver'
+require 'rspec'
+
+describe Solver do
+  describe '#factorial' do
+    it 'returns 1 for input 0' do
+      expect(Solver.new.factorial(0)).to eq(1)
+    end
+
+    it 'returns 120 for input 5' do
+      expect(Solver.new.factorial(5)).to eq(120)
+    end
+
+    it 'raises an exception for a negative integer' do
+      expect { Solver.new.factorial(-1) }.to raise_error(ArgumentError)
+    end
+  end
+
+  describe '#reverse' do
+    it 'returns "olleh" for input "hello"' do
+      expect(Solver.new.reverse('hello')).to eq('olleh')
+    end
+
+    it 'returns "" for an empty string' do
+      expect(Solver.new.reverse('')).to eq('')
+    end
+
+    it 'returns the same character for a single character string' do
+      expect(Solver.new.reverse('a')).to eq('a')
+    end
+    describe '#fizzbuzz' do
+      it 'returns Fizzbuzz when given 15' do
+        expect(Solver.new.fizzbuzz(15)).to eq('fizzbuzz')
+      end
+      it 'returns Fizz when given 3' do
+        expect(Solver.new.fizzbuzz(3)).to eq('fizz')
+      end
+      it 'returns Buzz when given 5' do
+        expect(Solver.new.fizzbuzz(5)).to eq('buzz')
+      end
+    end
```

Comment:
- [ ] In TDD tests are supposed to cover all edge cases, As per this [requirement](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/testing/tdd_project.md#:~:text=Any%20other%20case%2C%20return%20N%20as%20a%20string%20(e.g.%20say%20N%20is%207%20then%20return%20%227%22)), Please add a test in case of a regular number like 7 `not divisible by 3 or 5` the method suppose to return it as it is.

Reply:
@Omar-Muhamad  Thanks for your kind feedback. Working on it.